src/Pagination/index.jsx

```diff
@@ -0,0 +1,414 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+
+import Button from '../Button';
+import Icon from '../Icon';
+import { ExtraSmall, LargerThanExtraSmall } from '../Responsive';
+import newId from '../utils/newId';
+import getTextFromElement from '../utils/getTextFromElement';
+
+import styles from './Pagination.scss';
+
+class Pagination extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.previousButtonRef = null;
+    this.nextButtonRef = null;
+
+    this.pageRefs = {};
+
+    this.state = {
+      currentPage: this.props.currentPage,
+      previousNextButtonClicked: false,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.currentPage !== this.props.currentPage) {
+      this.state.currentPage = nextProps.currentPage;
+    }
+  }
+
+  componentDidUpdate() {
+    const { currentPage, previousNextButtonClicked } = this.state;
+    const currentPageRef = this.pageRefs[currentPage];
+
+    if (currentPageRef && !previousNextButtonClicked) {
+      currentPageRef.focus();
+    }
+  }
+
+  handlePageSelect(page) {
+    if (page !== this.state.currentPage) {
+      this.setState({
+        currentPage: page,
+        previousNextButtonClicked: false,
+      });
+      this.props.onPageSelect(page);
+    }
+  }
+
+  handlePreviousNextButtonClick(page) {
+    const { pageCount } = this.props;
+
+    this.setState({
+      currentPage: page,
+      previousNextButtonClicked: true,
+    });
+
+    if (page === 1) {
+      this.nextButtonRef.focus();
+    } else if (page === pageCount) {
+      this.previousButtonRef.focus();
+    }
+
+    this.props.onPageSelect(page);
+  }
+
+  renderEllipsisButton() {
+    return (
+      <li
+        className={classNames([styles['page-item'], styles.disabled])}
+        key={newId('pagination-ellipsis-')}
+      >
+        <span
```

Comment:
Haha, I actually had this as a Button component before but changed it to a `span` after @georgebabey's suggestion. His rationale for `span` over `Button` was because the `Button` is always disabled, so it's never clickable. I was using the `Button` before solely for the styles but the `.btn` class on the `span` works just as well. I think it makes more semantic sense to keep it as a `span` since the `Button` would always be disabled.